overlord/devicestate/devicestate_test.go

```diff
@@ -3265,6 +3270,246 @@ func (s *deviceMgrSuite) TestDoRequestSerialReregistration(c *C) {
 	c.Assert(err, IsNil)
 }
 
+func (s *deviceMgrSuite) TestDoRequestSerialReregistrationStreamFromService(c *C) {
+	mockServer := s.mockServer(c, "REQID-1", nil)
+	defer mockServer.Close()
+
+	restore := devicestate.MockBaseStoreURL(mockServer.URL)
+	defer restore()
+
+	s.ancillary = s.brands.AccountsAndKeys("rereg-brand")
```

Comment:
AIUI this is one of the key pieces of the test so maybe a small comment to highlight this would be good. Maybe something like: `// provide stream of assertion when re-registering` (or whatever is most apporiate).

Related, this test is almost identical to TestDoRequestSerialReregistration and TestDoRequestSerialReregistrationIncompleteStreamFromService maybe there is some room for consolidation? Maybe something like: http://paste.ubuntu.com/p/QW2qrWg7vs/ ? and if we find a way to include TestDoRequestSerialReregistrationDoubleSerialStreamFromService that would be even better I guess.

Reply:
I did a bit of that, I didn't share the tail with the checks though